test/fuzz/transformation_changing_memory_semantics_test.cpp

```diff
@@ -0,0 +1,482 @@
+  const auto context = BuildModule(env, consumer, shader, kFuzzAssembleOption);
+  spvtools::ValidatorOptions validator_options;
+  ASSERT_TRUE(fuzzerutil::IsValidAndWellFormed(context.get(), validator_options,
+                                               kConsoleMessageConsumer));
+  TransformationContext transformation_context(
+      MakeUnique<FactManager>(context.get()), validator_options);
+
+  // Bad: Instruction does not exist.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(150, SpvOpAtomicLoad, 0), 0, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+#ifndef NDEBUG
+
+  // Bad: Operand index does not equal 0 or 1.
+  ASSERT_DEATH(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 2, 100)
+                   .IsApplicable(context.get(), transformation_context),
+               "The instruction may not be an atomic or barrier instruction. \
+                The operands index may be not equal 0 or 1. \
+                The index may be equal to one and the expected is zero.");
+
+  // Bad: The SpvOpAtomicLoad takes one memory semantic value and the operand
+  // index passed is 1.
+  ASSERT_DEATH(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 1, 100)
+                   .IsApplicable(context.get(), transformation_context),
+               "The instruction may not be an atomic or barrier instruction. \
+                The operands index may be not equal 0 or 1. \
+                The index may be equal to one and the expected is zero.");
+#endif
+  // Bad: The new Id instruction does not exist.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 0, 101)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The new Id instruction exists, but not OpConstant.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(21, SpvOpAtomicLoad, 0), 0, 13)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The OpAtomicStore memory semantics value cannot be AcquireRelease.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(24, SpvOpAtomicStore, 0), 0, 100)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The OpAtomicExchange(read-modify-write) memory semantics value cannot
+  // be changed from Acquire to Release.
+  ASSERT_FALSE(TransformationChangingMemorySemantics(
+                   MakeInstructionDescriptor(23, SpvOpAtomicExchange, 0), 0, 99)
+                   .IsApplicable(context.get(), transformation_context));
+
+  // Bad: The SpvOpAtomicCompareExchange(read-modify-write) takes two memory
+  // semantics, unequal can't acquire/release value.
```

Comment:
This comment is very hard to understand because you use "unequal" without saying what it means or what it is. You should introduce the two operands, or at least mention that it is an operand (e.g. "the unequal operand") because then at least the reader knows that they need to lookup the operands for `OpAtomicCompareExchange`.

I suggest that you say "takes two memory semantics operands: Equal and Unequal. Unequal can't be...". Or "takes two memory semantics operands. The second operand cannot be ...". 

Also, I think the comment is incorrect. I believe the Unequal operand must not be `Release` nor `AcquireRelease`. So I guess you meant to say `AcquireRelease` instead of `acquire/release`. In English, `/` usually means "either", so either `Acquire` or `Release`, which is different to the `AcquireRelease` value (which means both acquire and release).

Reply:
Improved.